RFCs/0011-DeletionPolicy.md

```diff
@@ -0,0 +1,262 @@
+            "Type": "AWS::DynamoDB::Table",
+            "Properties": {
+              "KeySchema": [
+               {
+                 "AttributeName" : "primaryKey",
+                 "KeyType" : HASH
+               } 
+              ]
+            },
+            "DeletionPolicy": { 
+                "Fn::If": [
+                    "IsProd",
+                    "Retain",
+                    "Delete"
+                ]
+            },
+            "UpdateReplacePolicy": { 
+                "Fn::If": [
+                    "IsProd",
+                    "Retain",
+                    "Delete"
+                ]
+            }
+        }
+    }
+}
+```
+
+### Use [Fn::FindInMap](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap.html) and `AWS::Region` Pseudo-Parameter
+```
+{
+    "AWSTemplateFormatVersion" : "2010-09-09",
+    "Parameters": {
+        "Stage": {
+            "Type": "String",
+            "AllowedValues": ["Prod", "Staging", "Dev"]
+        }
+    },
+    "Mappings" : {
+        "RegionMap" : {
+            "us-east-1" : { 
+                "DeletionPolicy": "Retain",
+                "UpdateReplacePolicy": "Retain"
+            },
+            "us-west-1" : { 
+                "DeletionPolicy": "Delete",
+                "UpdateReplacePolicy": "Delete"
+            },
+            "eu-west-1" : { 
+                "DeletionPolicy": "Retain",
+                "UpdateReplacePolicy": "Delete"
+            },
+            "ap-southeast-1" : { 
+                "DeletionPolicy": "Delete",
+                "UpdateReplacePolicy": "Delete"
+            }
+        }
+    },
+    "Resources": {
+        "Bucket": {
+            "Type": "AWS::S3::Bucket",
+            "DeletionPolicy": { 
+                "Fn::FindInMap" : [
+                    "RegionMap", 
+                    { "Ref" : "AWS::Region" },
+                    "DeletionPolicy"
+                ]
+            },
+            "UpdateReplacePolicy": { 
+                "Fn::FindInMap" : [
+                    "RegionMap", 
+                    { "Ref" : "AWS::Region" },
+                    "UpdateReplacePolicy"
+                ]
+            }
+        }
+    }
+}
+```
+
+# Motivation
+
+CloudFormation customers expect to use conditions and intrinsic functions as a value for a `DeletionPolicy` and `UpdateReplacePolicy` Resource attributes as they can do elsewhere in the template.
+Main ask from our customers was to allow them to vary DeletionPolicy and UpdateReplacePolicy values based on development stage, which tends to be passed in as a parameter, e.g., allow resource deletion in Dev stages but not Prod. This RFC proposes to support not only such use case but also other intrinsic function usages and Pseudo-parameter references.
+
+* References
+    * https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/162 (148 thumbs up)
+    * https://github.com/aws/aws-cli/issues/3825
```

Comment:
Not a doc change, but you should add a comment to this issue linking them to issue 58 in this repo.